weave/flow/chat_util.py

```diff
@@ -22,75 +22,83 @@ def __init__(self, chunk_iter: Iterator[ChatCompletionChunk]) -> None:
-                for i in range(chunk_choice.index + 1 - len(self.output_choices)):
-                    self.output_choices.append(
-                        {
-                            "index": len(self.output_choices),
-                            "message": {
-                                "content": None,
-                                "tool_calls": None,
-                            },
-                            "finish_reason": None,
-                            "logprobs": None,
-                        }
-                    )
 
-                # choice fields
-                choice = self.output_choices[chunk_choice.index]
-                if chunk_choice.finish_reason:
-                    choice["finish_reason"] = chunk_choice.finish_reason
-                if chunk_choice.logprobs:
-                    choice["logprobs"] = chunk_choice.logprobs
+    def _process_chunk(self, chunk: ChatCompletionChunk) -> None:
+        if self.first_chunk is None:
+            self.first_chunk = chunk
+        for chunk_choice in chunk.choices:
+            for i in range(chunk_choice.index + 1 - len(self.output_choices)):
+                self.output_choices.append(
+                    {
+                        "index": len(self.output_choices),
+                        "message": {
+                            "content": None,
+                            "tool_calls": None,
+                        },
+                        "finish_reason": None,
+                        "logprobs": None,
+                    }
+                )
+
+            # choice fields
+            choice = self.output_choices[chunk_choice.index]
+            if chunk_choice.finish_reason:
+                choice["finish_reason"] = chunk_choice.finish_reason
+            if chunk_choice.logprobs:
+                choice["logprobs"] = chunk_choice.logprobs
 
-                # message
-                if chunk_choice.delta.content:
-                    if choice["message"]["content"] is None:
-                        choice["message"]["content"] = ""
-                    choice["message"]["content"] += chunk_choice.delta.content
-                if chunk_choice.delta.role:
-                    choice["message"]["role"] = chunk_choice.delta.role
+            # message
+            if chunk_choice.delta.content:
+                if choice["message"]["content"] is None:
+                    choice["message"]["content"] = ""
+                choice["message"]["content"] += chunk_choice.delta.content
+            if chunk_choice.delta.role:
+                choice["message"]["role"] = chunk_choice.delta.role
 
-                # function call
-                if chunk_choice.delta.function_call:
-                    raise NotImplementedError("Function calls not supported")
+            # function call
+            if chunk_choice.delta.function_call:
+                raise NotImplementedError("Function calls not supported")
```

Comment:
@tssweeney:  I would like to understand why function calls are not supported here. I know this is deprecated in Openai SDK in favor of tool calls, but libraries like LangChain currently still use this feature, and this causes the following test case related to agents break in LangChain: 

https://github.com/wandb/weave/blob/6cd218ec90f20264ef45598123f3e02e37aa11a0/weave/integrations/langchain/langchain_test.py#L305